src/Symfony/Component/Messenger/Transport/AmqpExt/AmqpReceiver.php

```diff
@@ -62,9 +63,14 @@ public function receive(callable $handler): void
 
                 throw $e;
             } catch (\Throwable $e) {
-                $this->connection->nack($AMQPEnvelope, AMQP_REQUEUE);
+                if (!$this->connection->publishForRetry($AMQPEnvelope)) {
```

Comment:
I think you need to handle errors here as well. Otherwise it kills the whole receiver.

Reply:
Added